File: Leetcode-one-by-one/src/Binary Watch.java
import java.util.*;

class Solution {
    public List<String> readBinaryWatch(int turnedOn) {
        List<String> result = new ArrayList<>();
        
        for (int hour = 0; hour < 12; hour++) {
            for (int minute = 0; minute < 60; minute++) {
                if (Integer.bitCount(hour) + Integer.bitCount(minute) == turnedOn) {
                    result.add(hour + ":" + String.format("%02d", minute));
                }
            }
        }
        
        return result;
    }
}

File: Leetcode-one-by-one/src/Binary Watch.swift
class Solution {
    func readBinaryWatch(_ turnedOn: Int) -> [String] {
        var result: [String] = []

        for hour in 0..<12 {
            for minute in 0..<60 {
                if bitCount(hour) + bitCount(minute) == turnedOn {
                    let time = "\(hour):" + String(format: "%02d", minute)
                    result.append(time)
                }
            }
        }

        return result
    }

    private func bitCount(_ n: Int) -> Int {
        var n = n
        var count = 0
        while n > 0 {
            count += n & 1
            n >>= 1
        }
        return count
    }
}
